includes/discovery/sensors/dbm/ciena.inc.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/*
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @author     Adam Kujawski <adamkuj@amplex.net>
+ */
+
+
+
+echo 'Ciena ';
+
+$multiplier = 1;
+$divisor    = 10000;
+foreach ($pre_cache['ciena_oids'] as $index => $entry) {
+    if (is_numeric($entry['wwpLeosPortXcvrRxDbmPower']) && $entry['wwpLeosPortXcvrRxDbmPower'] != 0 && $entry['wwpLeosPortXcvrTxDbmPower'] != 0) {
+        $oid = '.1.3.6.1.4.1.6141.2.60.4.1.1.1.1.105.'.$index;
+        $descr = dbFetchCell('SELECT `ifDescr` FROM `ports` WHERE `ifName`= ? AND `device_id` = ?', array($index, $device['device_id'])) . ' Rx Power';
```

Comment:
I'm never a fan of this as it relies on port discovery to have run first. Can we not get the port description from within the mib?